src/components/MessageActions/MessageActionsBox.tsx

```diff
@@ -46,42 +48,27 @@ const UnMemoizedMessageActionsBox = <
     isUserMuted,
     mine,
     open = false,
+    referenceElement,
   } = props;
 
   const {
     CustomMessageActionsList = DefaultCustomMessageActionsList,
   } = useComponentContext<StreamChatGenerics>('MessageActionsBox');
   const { setQuotedMessage } = useChannelActionContext<StreamChatGenerics>('MessageActionsBox');
-  const { customMessageActions, message, messageListRect } = useMessageContext<StreamChatGenerics>(
+  const { customMessageActions, message } = useMessageContext<StreamChatGenerics>(
     'MessageActionsBox',
   );
 
   const { t } = useTranslationContext('MessageActionsBox');
 
-  const [reverse, setReverse] = useState(false);
+  const { attributes, popperElementRef, styles } = useMessageActionsBoxPopper<HTMLDivElement>({
```

Comment:
I am thinking whether these could be moved up to `MessageActions` and `MessageActionsBox` would only get `ComponentProps<'div'>`. I am concerned about new prop `referenceElement` being introduced only for popper to work, but the integrators using this component may not know what to do with it. Also `referenceElement` is not mandatory, so that would be a breaking change potentially? I am thinking whether `MessageActionsBox` could forwardRef and accept div props that would be spread to the root of `MessageActionsBox`.

Not sure what @arnautov-anton or @oliverlaz think.

When adding a new prop we should document

Reply:
I think your proposal makes sense (although it would make the implementation less clean). Let's try it.

Reply:
@MartinCupela Thanks for the suggestion! I agree we should try to avoid adding a new mandatory prop, and moving the logic up to the `MessageActions` makes sense.

Reply:
@MartinCupela Fixed in https://github.com/GetStream/stream-chat-react/pull/2241/commits/8b30c038057e4cd9aee772eee6a858aeb8189ade.

However, instead of spreading div props on `MessageActionsBox` I just wrapped in an other div that is used solely for positioning - that felt a bit cleaner to me.